### PR TITLE
PR: Remove ujson dependency for Windows (it requires the install of MSVC compiler)

### LINKED ISSUE #10763: Fails to install Spyder4.0.0rc2 on Windows due to ujson dependency
## Problem Description
First of all, this is a dependency issue so it may be not solved by the Spyder-team but I thought that you may want to know about it.

Spyder 4.0.0rc2 actually fails to install (via pip) on Windows if MSVC compiler is not installed. This is related to the fact that some dependencies depend on `ujson` package, which require MSVC 14 as shown in the log below.

So the dependency tree goes like that: rc2 requires `python-language-server` 0.31.0, which requires `ujson` and `python-jsonrpc-server` 0.3.0, which requires `ujson` too:
```
spyder 4.0.0rc2
    python-language-server 0.31.0
        ujson (since python-language-server 0.31.0, previous 0.30.0)
        python-jsonrpc-server 0.3.0
            ujson (since python-jsonrpc-server 0.3.0, previous 0.2.0)
```

This is working with rc1 when pre-installing first python-jsonrpc-server v0.2.0, then python-language-server 0.30.0.

### What steps reproduce the problem?

1. Install release candidate 2 on W10  `pip install --pre spyder`

### Error

```
[...]
  Running setup.py install for ujson ... error
[...]
    Warning: 'classifiers' should be a list, got type 'filter'
    running install
    running build
    running build_ext
    building 'ujson' extension
    error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": https://visualstudio.microsoft.com/downloads/
    ----------------------------------------
```

## Versions

* Spyder version: -
* Python version: 3.7.5
* Qt version: -
* PyQt version: - 
* Operating System name/version: Windows 10, AMD64

### PR DESCRIPTION
**Pull request by dalthviz**
As discussed with @ccordoba12
Fixes spyder-ide/spyder#10763

### TODO:
 - [x] Fix tests

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/python-jsonrpc-server](https://github.com/palantir/python-jsonrpc-server), @dalthviz! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: setup.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
     install_requires=[
         'future>=0.14.0',
         'futures; python_version<"3.2"',
-        'ujson<=1.35',
+        'ujson<=1.35;platform_system=="Linux"',
```
Comment (ccordoba12) [inline setup.py]: Please change this to `platform_system!="Windows"`

File: examples/langserver_ext.py [hunk 2]
```diff
@@ -7,6 +6,11 @@
 
 from pyls_jsonrpc import streams
 
+try:
+    import ujson as json
+except ImportError:
```
Comment (ccordoba12) [inline examples/langserver_ext.py]: Please change this `ImportError` to catch all possible errors generated when importing `ujson`. The thing is more complex errors can be thrown for compiled dependencies.

**Comment by ccordoba12** [comment]
@gatesn, I think this is ready. We need this to allow Windows users to install Spyder with pip.

Could you merge it and release a new version (0.3.2) with it? Thanks!

**Comment by ccordoba12** [comment]
@jroitgrund, it seems you have write permissions in this repo. Could you help us to merge this PR?

That way I could create a new release of `python-language-server` (I'm a maintainer of that project) that takes into account this and #28.

**Review by jroitgrund — APPROVED** [review]

**Comment by ccordoba12** [comment]
Thanks a lot @jroitgrund! Could you release `0.3.2` with this fix? Thanks!

**Pull request merged**